app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/ResurveyOpeningHours.kt

```diff
@@ -0,0 +1,78 @@
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.DateUtil
+
+class ResurveyOpeningHours(private val overpassServer: OverpassMapDataDao) : OsmElementQuestType<OpeningHoursAnswer> {
+    override val commitMessage = "resurvey opening hours"
+    override val icon = R.drawable.ic_quest_opening_hours
+    override fun getTitle(tags: Map<String, String>) = R.string.resurvey_opening_hours_title
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        return arrayOf(tags["name"]!!)
+    }
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox)) { element, geometry ->
+            if (element.tags != null) {
+                // require opening hours that are supported
+                if (OpeningHoursTagParser.parse(element.tags["opening_hours"]!!) != null) {
+                    handler.handle(element, geometry)
+                } else {
+                    Log.wtf("AAAA", "opening_hours=" + element.tags["opening_hours"] + " was rejected as not representable in SC")
+                }
+            }
+
+        }
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox): String {
+        val groupName = ".old_opening_hours"
+        val nodeGroupName = groupName + "_nodes"
+        val wayGroupName = groupName + "_ways"
+        val relationGroupName = groupName + "_relations"
+        val reviewIntervalInDays = 380
+        return bbox.toGlobalOverpassBBox() + "\n" +
+                getQueryPart("node", nodeGroupName, reviewIntervalInDays) +
+                getQueryPart("way", wayGroupName, reviewIntervalInDays) +
+                getQueryPart("relation", relationGroupName, reviewIntervalInDays) +
+                "($nodeGroupName; $wayGroupName; $relationGroupName;);\n" +
+                getQuestPrintStatement()
+    }
+
+    // Note that newer segment will ensure that any edit,
+    // including adding or updating review marker like check_date or survey:date tags
+    // will cause OSM elements to become ineligible for this quest for reviewIntervalInDays days.
+    // It allows supporting check_date and any other survey markers without parsing of any tags.
+    //
+    // objects where tag was surveyed and confirmed to be correct by adding check_date tags
+    // are with larger delay, as it means this info on this specific object is relatively stable
+    protected fun getQueryPart(objectType: String, nameOfGeneratedGroup: String, reviewIntervalInDays: Int) =
+            "$objectType[name][opening_hours]['opening_hours:signed'!='no'](if:!is_date(t['check_date:opening_hours']) || date(t['check_date:opening_hours']) < " +
```

Comment:
The query should also take into account the other alternatives of check_date, if there are (still) any around.

Reply:
It is checking just `check_date:opening_hours` ( https://taginfo.openstreetmap.org/keys/check_date%3Aopening_hours#overview )

For other variants see https://taginfo.openstreetmap.org//search?q=opening_hours

There is `opening_hours:check_date` with just 329 uses (on the other hand, it is 1/3 of `check_date:opening_hours`). But I think it is OK to not support this.

Later things with 15 uses and less - see say https://taginfo.openstreetmap.org/keys/survey%3Adate%3Aopening_hours

Reply:
There is `opening_hours:lastcheck` with 1.6k uses.
I agree `check_date` is the more widely adapted tag and is also documented as opposed to `lastcheck`, but spread of `lastcheck` will not vanish over night, so that should IMO be supported.

Reply:
I added `changes.deleteIfExists("opening_hours:lastcheck")` on solving.

In query I think it may be skipped. Edit adding `opening_hours:lastcheck` will update timestamp (like any edit), so for 380 days resurvey quest will not be created.

But 1080 days of resurvey delay done for `check_date:opening_hours` should be OK to skip as it is below 2000 uses worldwide and increase to query complexity seems not worth it .